versions.go

```diff
@@ -161,6 +160,7 @@ func createNewMongoDBVersion(parsedVersion LegacyMongoDBVersion) (*NewMongoDBVer
 	}
 	v.quarter = v.String()[:3]
 	if strings.Contains(v.tag, devReleaseTag) {
+		v.isDev = false
```

Comment:
This should be false for all `NewMongoDBVersion`s, not just development releases.

Reply:
It is allowed in the new versioning scheme, but we are just not expecting it to be there. But in the case that it does, we should expect the new versioning scheme to return true for `IsDevelopmentBuild`.

Reply:
It shouldn't be allowed because "pre" and versions ending in "-" are not allowed in the new scheme. There's no case in the new versioning scheme when `IsDevelopmentBuild` returns true.

Reply:
I didn't realize that 🤔  what about ~ ?

Reply:
I think those are counted the same as dev commits/builds (for some reason, this repo calls it dev build but mr calls it a dev commit). I believe that no dev builds have ever actually been released so far. At least, if you do a `git tag -l` on the mongo repo, there's no sign that any dev commit has ever actually been made and released.

Since we're only supporting the "alpha" syntax for development releases, I think it's safe to say that the dev commit thing was a theoretical possibility under the legacy scheme but won't exist in the new scheme.

Reply:
Should we error then if we get something with `pre`?

Reply:
That seems reasonable - it would be nice to limit the set of acceptable inputs.

Reply:
Added error below.